### PR TITLE
BART Database Integration

### PR DESCRIPTION
**Pull request by TaxiKab119**
This PR implements the BART database and adds a new selection screen to select trial details that form the basis of BART trial entities to write BART trial data into. It also ensures that when a practice trial is selected that no data is written to the database.

### CODE REVIEW

File: app/src/main/java/com/example/dancognitionapp/AppViewModelProvider.kt [hunk 1]
```diff
@@ -21,6 +23,16 @@ object AppViewModelProvider {
                 danCognitionApplication().container.participantRepository
             )
         }
+        initializer {
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/AppViewModelProvider.kt:26]: One of my bigger comments is that we can leverage the `BartRepository` interface to create an empty `PracticeBartRepository` implementation that just overrides the methods but with empty/returning null functions. We can create a function in here like
```
fun testViewModelFactory(isPractice: Boolean)  = viewModelFactory {
        initializer {
            BartViewModel(
                if (isPractice) PracticeBartRepository() else danCognitionApplication().container.bartRepository
            )
        }
    }
```
that has initializers for both test viewmodels, and keep the plain `val factory` for everything else that depends on the `participantRepository`. This way, we don't have to keep track of `isPractice` in our UI state at all. Makes both the state and viewmodel cleaner and more maintainable.

File: app/src/main/java/com/example/dancognitionapp/assessment/bart/ui/BartFragment.kt [hunk 2]
```diff
@@ -5,28 +5,72 @@ import android.view.LayoutInflater
...
                     findNavController().popBackStack(destination, inclusive = false)
                 }
             }
         }
         return view
     }
-}
\ No newline at end of file
+    private fun initializeBart(isPractice: Boolean) {
+        if (isPractice) {
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/assessment/bart/ui/BartFragment.kt:60]: As per my comment above, we won't need this check anymore and can simply just `initBart` with the args, as none of that will matter when the viewmodel has a `PracticeBartRepository`

File: app/src/main/java/com/example/dancognitionapp/assessment/bart/ui/BartTestScreen.kt [hunk 3]
```diff
@@ -184,10 +181,10 @@ fun BartButton(
...
 
-@LandscapePreview
-@Composable
-fun BartConstraintLayoutPreview() {
-    DanCognitionAppTheme {
-        BartTestScreen(modifier = Modifier.fillMaxSize(), {})
-    }
-}
\ No newline at end of file
+//@LandscapePreview
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/assessment/bart/ui/BartTestScreen.kt:184]: remove commented-out code

File: app/src/main/java/com/example/dancognitionapp/assessment/selection/SelectionFragment.kt [hunk 4]
```diff
@@ -34,21 +37,37 @@ class SelectionFragment: Fragment() {
...
             }
         }
         return view
     }
 }
 
-@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun TestSelectScreen(modifier: Modifier = Modifier, isPractice: Boolean = false, onClick: (Int) -> Unit = {}) {
+fun PracticeTestSelectScreen(modifier: Modifier = Modifier, isPractice: Boolean = false, onClick: (Int) -> Unit = {}) {
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/assessment/selection/SelectionFragment.kt:70]: Each parameter on new line here, and remove `isPractice` as it's not used

File: app/src/main/java/com/example/dancognitionapp/assessment/selection/SelectionFragment.kt [hunk 5]
```diff
@@ -70,54 +88,26 @@ fun TestSelectScreen(modifier: Modifier = Modifier, isPractice: Boolean = false,
 }
 
 @Composable
-fun SelectionPageContent(isPractice: Boolean, modifier: Modifier = Modifier, onClick: (Int) -> Unit = {}) {
+fun PracticeSelectionPageContent(modifier: Modifier = Modifier, onClick: (Int) -> Unit = {}) {
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/assessment/selection/SelectionFragment.kt:91]: new line for each param

File: app/src/main/java/com/example/dancognitionapp/assessment/selection/SelectTestDetailsScreen.kt [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SelectionDropDownMenu(
+    @StringRes titleRes: Int,
+    @StringRes labelRes: Int,
+    menuOptions: List<String>,
+    modifier: Modifier = Modifier,
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/assessment/selection/SelectTestDetailsScreen.kt:148]: modifier unused

File: app/src/main/java/com/example/dancognitionapp/AppViewModelProvider.kt (hunk 1 above)
Comment (TaxiKab119) [reply app/src/main/java/com/example/dancognitionapp/AppViewModelProvider.kt:26]: This is so great I love it

**Comment by TaxiKab119** [comment]
Hey, just based on the nature of how I continued with the NBack DB branch off of this branch, I just combined both into the most recent PR.

**Pull request merged**